druid/src/env.rs

```diff
@@ -371,6 +384,21 @@ impl Env {
         let color_num = id as usize % self.0.debug_colors.len();
         self.0.debug_colors[color_num].clone()
     }
+
+    /// Replaces and adds to current `Env`'s values with values form the provided `Env`.
+    pub(crate) fn with_overrides(&self, changes: &Env) -> Env {
+        let mut new_env = self.to_owned();
+        let mut internal_env = Arc::make_mut(&mut new_env.0).to_owned();
```

Comment:
we shouldn't need to do the `to_owned`, we can just use the mutable reference directly?

Reply:
which mutable reference? I can't access any of them because self isn't `mut`.

Reply:
I guess there are maybe two problems here:

- `Env` has a weird internal layout. It wasn't really designed with this particular pattern (overrides from another env) in mind. The rationale for `EnvImpl` is just to make sure that cloning an `Env` requires a single pointer clone. There's the additional problem that the `Env` is sort of conceptually vague; to the user the `Env` is just a key/value store, but it also happens to contain localization info etc.
- This is not helped by the fact that the API we've designed here represents the overrides as a new `Env`. The issue here is that each new `Env` is going to contain a duplicate version of the localization manager, and this is silly.

Some possible solutions:

- make the localization stuff something that has to be set up explicitly, and `Env::default()` just has no localization info.
- have a separate `EnvMap` type that is used for specifying the overrides

I don't *love* having a new type, since it's another import, another piece of public api, etc; I think it might make more sense to move to having the localization setup be something that you have to do manually, and which druid can do when setting up the initial env...


```rust
let mut new_env = self.clone();
let internal = Arc::make_mut(&mut new_env);

Reply:
Hmm I tried doing what you suggested but the borrow checker doesn't allow that. Maybe I'm approaching it the wrong way. I'm quite rusty on rust right now lol.